lib/Storage/MySqlTimeSeriesStore.php

```diff
@@ -0,0 +1,371 @@
+            $entry['maxEnd'] = $row['MaxEnd'];
+            $entry['layoutId'] = $row['layoutId'];
+            $entry['widgetId'] = $row['widgetId'];
+            $entry['mediaId'] = $row['mediaId'];
+
+            $rows[] = $entry;
+        }
+
+        // Paging
+        $results = [];
+        if ($limit != '' && count($rows) > 0) {
+            $results = $this->store->select('
+              SELECT COUNT(*) AS total FROM (SELECT stat.type, display.Display, layout.Layout, IFNULL(`media`.name, IFNULL(`widgetoption`.value, `widget`.type)) ' . $body . ') total
+            ', $params);
+        }
+
+        return [
+            'statData' => $rows,
+            'count' => count($rows),
+            'totalStats' => $results[0]['total']
+        ];
+
+    }
+
+    /** @inheritdoc */
+    public function getEarliestDate()
+    {
+        $earliestDate = $this->store->select('SELECT MIN(statDate) AS minDate FROM `stat`', []);
+
+        return $earliestDate;
+    }
+
+    /** @inheritdoc */
+    public function getStats($fromDt, $toDt, $displayIds = null)
+    {
+        $sql = '
+        SELECT stat.*, display.Display, layout.Layout, media.Name AS MediaName
+          FROM stat
+            INNER JOIN display
+            ON stat.DisplayID = display.DisplayID
+            LEFT OUTER JOIN layout
+            ON layout.LayoutID = stat.LayoutID
+            LEFT OUTER JOIN media
+            ON media.mediaID = stat.mediaID
+         WHERE 1 = 1
+          AND stat.end > :fromDt
+          AND stat.start <= :toDt
+          
+        ';
+
+        if(count($displayIds) > 0) {
+            $sql .= ' AND stat.displayID IN (' . implode(',', $displayIds) . ')';
+        }
+
+        $params = [
+            'fromDt' => $fromDt,
+            'toDt' => $toDt
+        ];
+
+        $sql .= " ORDER BY stat.start ";
+
+        // Run our query using a connection object (to save memory)
+        $connection = $this->store->getConnection();
+        $statement = $connection->prepare($sql);
+
+        // Execute
+        $statement->execute($params);
+
+        $rows = [];
+
+        while ($row = $statement->fetch(\PDO::FETCH_ASSOC)) {
+
+            $entry = [];
+            // Read the columns
+            $entry['type'] = $row['type'];
+            $entry['start'] = $row['start'];
+            $entry['end'] = $row['end'];
+            $entry['layout'] = $row['Layout'];
+            $entry['display'] = $row['Display'];
+            $entry['media'] = $row['MediaName'];
+            $entry['tag'] = $row['tag'];
+            $entry['displayId'] = $row['displayId'];
+            $entry['layoutId'] = $row['layoutId'];
+            $entry['widgetId'] = $row['widgetId'];
+            $entry['mediaId'] = $row['mediaId'];
+
+            $rows[] = $entry;
+
+        }
+
+        return ['statData'=> $rows];
+
+    }
+
+    /** @inheritdoc */
+    public function deleteStats($maxage, $fromDt = null, $options = [])
+    {
+        try {
```

Comment:
Here I think we should specify some default options (considering `$options` is null). Something like:

```
$options = array_merge([
  'limit' => 10000,
   /*etc*/
], $options);```

Reply:
I am not sure what happens if you pass `null` into `array_merge` ?